### ISSUE TITLE
Rename the word "ingesters" in the "ring" package 

### ISSUE THREAD

**Issue by alvinlin123** [cortexproject/cortex#2723]
It seems like the ring package was originally written to be used by ingester only, so the word "ingester" is used in a lot of places which is quite confusing for new comer like me. I was reading [the ruler code], and wondered why ruler talks to ingester to get all sharded rules :)

It's going to be a big refactor (hence risky), but maybe we can break it into smaller steps? For example we can start by renaming just field, variable, and comments?

**Comment by pracucci** [comment]
Yes, we should do it. I also agree on doing it as incremental steps. Are you willing to give it a try?

**Comment by alvinlin123** [comment]
Yea I am welling to :)

**Comment by pracucci** [comment]
_Still valid, help wanted!_

**Comment by pracucci** [comment]
Some of the renaming has already been done (from "ingester" to "instance") but we still have protobuf and few variables/comments to update.

_Help wanted, thanks!_

**Comment by bboreham** [comment]
Some progress: #3762
Still more to do.
